src/typed-error.ts

```diff
@@ -23,20 +15,20 @@ if (Error.captureStackTrace != null) {
 	};
 }
 
-class TypedError extends BaseError {
+export class TypedError extends Error {
 	public name: string;
 	public message: string;
-	public stack: string;
+	public stack: string = '';
+
 	constructor(err: Error | string = '') {
 		super();
 		if (err instanceof Error) {
 			this.message = err.message;
 		} else {
 			this.message = err;
 		}
+		Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
What does this do?

Reply:
[see here](https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#support-for-newtarget).

Reply:
:+1: